File: VacationAppBackEnd/Models/VacationRequest.cs
using VacationAppBackEnd.Enums;

namespace VacationAppBackEnd.Models
{
    public class VacationRequest
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public User RequestedBy { get; set; } = null!;
        public DateOnly StartDate { get; set; }
        public DateOnly EndDate { get; set; }
        public VacationRequestStatus Status { get; set; }
        public string? Comment { get; set; } 
        public DateTime CreatedAt { get; set; }
        public int? ReviewedById { get; set; }
        public User? ReviewedBy { get; set; }
        public DateTime? DecisionDate { get; set; }
    }
}

File: VacationAppFrontEnd/src/types/vacation-request.ts
interface VacationRequest {
  id: number;
  requestedBy: {
    email: string;
    fullName: string;
    id: number;
  };
  startDate: string;
  endDate: string;
  type: string;
  status: string;
  comment?: string;
  createdAt: Date;
  reviewedBy: {} | null;
  decisionDate: Date | null;
}
